### PR TITLE
Fixed Mali Performance In Open GL and Also Fixed Compile

### PR DESCRIPTION
**Pull request by Ishan0981**
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7141)

### CODE REVIEW

**Comment by GPUCode** [comment]
The code still does not compile, you need to add the is_mali_gpu member variable

**Comment by Ishan0981** [comment]
Ok Wait

File: src/video_core/renderer_opengl/gl_rasterizer.cpp [hunk 1]
```diff
@@ -20,6 +20,12 @@
 
 namespace OpenGL {
 
+static bool IsVendorMali() {
```
Comment (SachinVin) [inline src/video_core/renderer_opengl/gl_rasterizer.cpp:23]: Should be `IsVendorArm`, Also this should not be here, it should be in gl_driver

File: src/video_core/renderer_opengl/gl_rasterizer.cpp [hunk 2]
```diff
@@ -82,13 +88,13 @@ RasterizerOpenGL::RasterizerOpenGL(Memory::MemorySystem& memory,
...
-      texture_buffer_size{TextureBufferSize()}, vertex_buffer{driver, GL_ARRAY_BUFFER,
+      texture_buffer_size{TextureBufferSize()},is_mali_gpu(IsVendorMali()), shader_dirty(true), vertex_buffer{driver, GL_ARRAY_BUFFER,
                                                               VERTEX_BUFFER_SIZE},
       uniform_buffer{driver, GL_UNIFORM_BUFFER, UNIFORM_BUFFER_SIZE},
       index_buffer{driver, GL_ELEMENT_ARRAY_BUFFER, INDEX_BUFFER_SIZE},
-      texture_buffer{driver, GL_TEXTURE_BUFFER, texture_buffer_size}, texture_lf_buffer{
+      texture_buffer{driver, GL_TEXTURE_BUFFER, is_mali_gpu ? 11264 : TEXTURE_BUFFER_SIZE}, texture_lf_buffer{
                                                                           driver, GL_TEXTURE_BUFFER,
-                                                                          texture_buffer_size} {
+                                                                          is_mali_gpu ? 525312 : TEXTURE_BUFFER_SIZE} {
```
Comment (SachinVin) [inline src/video_core/renderer_opengl/gl_rasterizer.cpp:97]: What is this magic number?

File: src/video_core/renderer_opengl/gl_rasterizer.cpp [hunk 3]
```diff
@@ -82,13 +88,13 @@ RasterizerOpenGL::RasterizerOpenGL(Memory::MemorySystem& memory,
     : VideoCore::RasterizerAccelerated{memory}, driver{driver_},
       shader_manager{renderer.GetRenderWindow(), driver, !driver.IsOpenGLES()},
       runtime{driver, renderer}, res_cache{memory, custom_tex_manager, runtime, regs, renderer},
-      texture_buffer_size{TextureBufferSize()}, vertex_buffer{driver, GL_ARRAY_BUFFER,
+      texture_buffer_size{TextureBufferSize()},is_mali_gpu(IsVendorMali()), shader_dirty(true), vertex_buffer{driver, GL_ARRAY_BUFFER,
```
Comment (SachinVin) [inline src/video_core/renderer_opengl/gl_rasterizer.cpp:91]: What is shader_dirty?

**Review by SachinVin — CHANGES_REQUESTED** [review]

**Comment by SachinVin** [comment]
Have you tested the changes? How much of an improvement is it?

**Comment by innovation2013** [comment]
#6375 This PR seems better than what you are doing now！

**Comment by FearlessTobi** [comment]
Can you also please give a clear explanation of your changes?
Why exactly does this improve performance?

**Comment by Ishan0981** [comment]
> The code still does not compile, you need to add the is_mali_gpu member variable

Maybe The Error Is Coming Because I Didn't Write This - #ifdef ARCHITECTURE_ARM64 And #include <string>  If You Can Tell Me Where The Error Is Coming Then I Will Be Helpful

**Comment by SachinVin** [comment]
This isn't the place for you to start learning C++.
It is inconsiderate and a waste of time to expect others to review changes that _you_ don't even know if they work.
Please do not open PRs if you don't understand what the changes are and what they do.

**Pull request closed without merging**